.github/workflows/test-generator.yml

```diff
@@ -0,0 +1,60 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Test Library Generation
+
+# Uses protos from HEAD of https://github.com/googleapis/google-cloudevents
+# and tooling from HEAD of this repository to update all generated code.
+#
+# It creates a PR whose branch will include an unmodified snapshot of the
+# tooling used to generate changes.
+
+on:
+  pull_request:
+    paths:
+      - 'tools/**'
+      - 'generate-code.sh'
+      - '.github/workflows/test-generator.yml'
+
+jobs:
+  # Create a pull request.
+  test-generator:
+    runs-on: ubuntu-latest
+    env:
+      GENERATE_DATA_SOURCE: protos/google-cloudevents
+
+    steps:
+    - name: Go Library > Checkout Repository
+      uses: actions/checkout@v3
+    
+    - name: Proto Schemas > Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        repository: googleapis/google-cloudevents
+        path: ${{ env.GENERATE_DATA_SOURCE }}
+
+    - name: Setup Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19.x
+
+    - name: Install the generator
+      run: |
+        bash tools/setup-generator.sh
+
+    - name: Run the generator
+      run: |
+        bash ./generate-code.sh
+      env:
+        GENERATE_PROTOC_PATH: tmp/protobuf/bin/protoc
```

Comment:
Future: this could be refactored into a composite action that both this workflow and generate workflow calls. See https://github.com/googleapis/google-cloudevents/pull/408/files for inspo

Reply:
I knew you'd say this ;)

I am undecided if I think 2 files with duplication is better than three DRY files, or if this is a GHA learning opportunity for later. Agreed it's future!